### DISCUSSION TITLE
How call an api on clicking on a button in refine.dev

### DISCUSSION THREAD

**Discussion by chinnabathini** [refinedev/refine#5718] [General]
whenever i click on button an api call should happen and also pass header to api. How to do this in refine.dev

**Comment by beg1c** [comment]
Hello @chinnabathini, it really depends upon data provider you are using. If you are new to all this, joining a Discord channel could be beneficial. There, you can interact with an AI bot that can promptly assist you with various questions and queries like these. It's a great way to get immediate guidance and support.

You can look at data hooks documentation on how to pass custom headers and adjust data provider if necessary:
https://refine.dev/docs/data/hooks/use-show/#usage
